trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/get-is-checkbox-checked.ts

```diff
@@ -0,0 +1,10 @@
+import { getFeeRecordIdsFromPremiumPaymentsCheckboxId } from '../../../helpers/premium-payments-table-checkbox-id-helper';
+import { PremiumPaymentsTableCheckboxId } from '../../../types/premium-payments-table-checkbox-id';
+
+export const getIsCheckboxChecked = (checkedFeeRecordIds: Set<number>): ((checkboxId: PremiumPaymentsTableCheckboxId) => boolean) => {
+  return (checkboxId: PremiumPaymentsTableCheckboxId): boolean => {
+    const checkboxFeeRecordIds = getFeeRecordIdsFromPremiumPaymentsCheckboxId(checkboxId);
+
+    return checkboxFeeRecordIds.some((feeRecordId) => checkedFeeRecordIds.has(feeRecordId));
+  };
+};
```

Comment:
I'm just wondering since the checkbox id generation is happening in the same mapping method as we set the isChecked-ness, the returned helper could just use a fee record id from the group rather than us reverse-engineering the ids from the checkbox id?

Reply:
Thanks, that's a nice optimisation spot